app/src/main/java/com/example/voyagerx/presenters/LaunchDetailsPresenter.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.voyagerx.presenters
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import androidx.core.content.ContextCompat.startActivity
+import com.example.voyagerx.LaunchDetailsFragment
+import com.example.voyagerx.R
+import com.example.voyagerx.repository.UserRepository
+import com.example.voyagerx.repository.model.Launch
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlin.coroutines.CoroutineContext
+
+class LaunchDetailsPresenter(private var userRepository: UserRepository, var view: LaunchDetailsFragment, private val context: Context) :
+    CoroutineScope
+{
+    private val job = Job() // keeps track of the state of a coroutine/cancel it
+    override val coroutineContext: CoroutineContext
```

Comment:
Nice, using a coroutine in the presenter was one weird part I ran into with using MVP.

Reply:
Yeah same. I'm still unclear on what exactly this code is doing haha